helm/templates/primary/replicaset-initiate-job.yaml

```diff
@@ -28,14 +28,84 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.namespace
+            {{- if eq .Values.config.security.authorization "enabled" }}
+            - name: MONGO_AUTHENTICATION_ENABLED
+              value: "true"
+            {{- end }}
+            {{- if or .Values.createUsers (eq .Values.config.security.authorization "enabled") }}
+            - name: MONGO_ROOT_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  {{- if .Values.useExistingSecret.enabled }}
+                  name: {{ .Values.useExistingSecret.name }}
+                  {{- else }}
+                  name: {{ template "mongodb.fullname" . }}
+                  {{- end }}
+                  key: mongodb-root-username
+            - name: MONGO_ROOT_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  {{- if .Values.useExistingSecret.enabled }}
+                  name: {{ .Values.useExistingSecret.name }}
+                  {{- else }}
+                  name: {{ template "mongodb.fullname" . }}
+                  {{- end }}
+                  key: mongodb-root-password
+            - name: MONGO_SERVICE_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  {{- if .Values.useExistingSecret.enabled }}
+                  name: {{ .Values.useExistingSecret.name }}
+                  {{- else }}
+                  name: {{ template "mongodb.fullname" . }}
+                  {{- end }}
+                  key: mongodb-service-username
+            - name: MONGO_SERVICE_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  {{- if .Values.useExistingSecret.enabled }}
+                  name: {{ .Values.useExistingSecret.name }}
+                  {{- else }}
+                  name: {{ template "mongodb.fullname" . }}
+                  {{- end }}
+                  key: mongodb-service-password
+            - name: MONGO_DEVELOPER_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  {{- if .Values.useExistingSecret.enabled }}
+                  name: {{ .Values.useExistingSecret.name }}
+                  {{- else }}
+                  name: {{ template "mongodb.fullname" . }}
+                  {{- end }}
+                  key: mongodb-developer-username
+            - name: MONGO_DEVELOPER_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  {{- if .Values.useExistingSecret.enabled }}
+                  name: {{ .Values.useExistingSecret.name }}
+                  {{- else }}
+                  name: {{ template "mongodb.fullname" . }}
+                  {{- end }}
+                  key: mongodb-developer-password
+            {{- end }}
           volumeMounts:
             - name: {{ include "mongodb.name" . }}-script-config
               mountPath: /scripts
+            {{- if .Values.createUsers }}
```

Comment:
```suggestion
            {{- if or .Values.createUsers (eq .Values.config.security.authorization "enabled") }}
```

Reply:
the scripts are not run if createUsers is false, so do we need to mount them? `mongo_create_users()` function in the configmap

Reply:
the check for ENV variables is because if authorization is enabled we need Root username and password for authenticating other requests